marklogic-data-hub-central/ui/src/components/entities/mapping/create-edit-mapping-dialog/create-edit-mapping-dialog.tsx

```diff
@@ -17,11 +17,14 @@ const CreateEditMappingDialog = (props) => {
   const [selectedSource, setSelectedSource] = useState(props.mapData && props.mapData != {} ? props.mapData.selectedSource : 'collection')
   const [srcQuery, setSrcQuery] = useState(props.mapData && props.mapData != {} ? props.mapData.sourceQuery : '');
   const [isQuerySelected, setIsQuerySelected] = useState(false);
+
   //To check submit validity
   const [isMapNameTouched, setMapNameTouched] = useState(false);
+  const [isNameMissingOnSave, setNameMissingOnSave] = useState(false);
   const [isDescriptionTouched, setDescriptionTouched] = useState(false);
   const [isCollectionsTouched, setCollectionsTouched] = useState(false);
   const [isSrcQueryTouched, setSrcQueryTouched] = useState(false);
+  const [isQueryMissingOnSave, setQueryMissingOnSave] = useState(false);
```

Comment:
Same suggestion as given for mapName earlier.

Reply:
Fixed!